src/main/java/Parser.java

```diff
@@ -0,0 +1,132 @@
+            throw new DukeException("Please specify a task number to mark! Correct usage: mark [Task Number]");
+        }
+
+        int taskIndex;
+
+        try {
+            taskIndex = Integer.parseInt(inputArray[1]);
+        } catch (NumberFormatException e) {
+            throw new DukeException("Task number must be in integer! Correct usage: mark [Task Number]");
+        }
+
+        if (taskIndex > taskList.taskList.size()) {
+            throw new DukeException("Please specify a task number that is within the list! "
+                    + "Correct usage: mark [Task Number]");
+        }
+        return new MarkCommand(taskIndex);
+    }
+
+    public static Command parseUnmarkTask(String input, TaskList taskList) throws DukeException {
+        String[] inputArray = input.split(" ");
+
+        if (inputArray.length < 2) {
+            throw new DukeException("Please specify a task number to mark! Correct usage: unmark [Task Number]");
+        }
+
+        int taskIndex;
+
+        try {
+            taskIndex = Integer.parseInt(inputArray[1]);
+        } catch (NumberFormatException e) {
+            throw new DukeException("Task number must be in integer! Correct usage: unmark [Task Number]");
+        }
+
+        if (taskIndex > taskList.taskList.size()) {
+            throw new DukeException("Please specify a task number that is within the list! "
+                    + "Correct usage: mark [Task Number]");
+        }
+
+        return new UnmarkCommand(taskIndex);
+    }
+
+    public static Command parseTodoTask(String input) throws DukeException {
+        if (input.split(" ").length < 2){
+            throw new DukeException("Your todo task cannot be empty! Correct usage: todo [Name of task]");
+        }
+        return new TodoCommand();
+    }
+
+    public static Command parseDeadlineTask(String input) throws DukeException {
+        String taskInformation = input.substring(BEGIN_DEADLINE_INDEX);
+        String[] taskInfoArray = taskInformation.split(" /by ");
+
+        if (taskInfoArray[0].trim().isEmpty()){
+            throw new DukeException("Your deadline task cannot be empty! "
+                    + "Correct usage: deadline [Name of task] /by [Deadline]");
+        }
+
+        if (taskInfoArray.length < 2){
+            throw new DukeException("You deadline task must include the '/by' keyword! "
+                    + "Correct usage: deadline [Name of task] /by [Deadline]");
+        }
+
+        if (taskInfoArray[1].trim().isEmpty()){
+            throw new DukeException("Your deadline cannot be empty! "
+                    + "Correct usage: deadline [Name of task] /by [Deadline]");
+        }
+
+        return new DeadlineCommand(taskInfoArray[0], taskInfoArray[1]);
+    }
+
+    public static Command parseEventTask(String input) throws DukeException {
+        String taskInformation = input.substring(BEGIN_EVENT_INDEX);
+        String[]fromSplitArray = taskInformation.split(" /from ");
```

Comment:
Maybe a space after []?
```suggestion
        String[] fromSplitArray = taskInformation.split(" /from ");
```